### ISSUE TITLE
Server adds erroneous nodes in response to filtered get request, beginning in 6ab0617

### ISSUE THREAD

**Issue by alangefe** [CESNET/Netopeer2#322]
To replicate, send a get request with the following subtree filter:

```
<filter xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <yang-library xmlns="urn:ietf:params:xml:ns:yang:ietf-yang-library">
    <module-set>
      <name>complete</name>
      <module>
        <name>ietf-yang-library</name>
        <revision/>
      </module>
    </module-set>
  </yang-library>
</filter>
```

The server should respond with the following subtree:

```
<data xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <yang-library xmlns="urn:ietf:params:xml:ns:yang:ietf-yang-library">
    <module-set>
      <name>complete</name>
      <module>
        <name>ietf-yang-library</name>
        <revision>2018-01-17</revision>
      </module>
    </module-set>
  </yang-library>
</data>
```

However, beginning in 6ab0617, the following malformed tree is returned:

```
<data xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <yang-library xmlns="urn:ietf:params:xml:ns:yang:ietf-yang-library">
    <module-set>
      <name>complete</name>
    </module-set>
    <module-set>
      <module>
        <name>ietf-yang-library</name>
      </module>
    </module-set>
    <module-set>
      <module>
        <revision>2018-01-17</revision>
      </module>
    </module-set>
  </yang-library>
</data>
```

Notice the erroneous extra termination nodes (`</module-set>` after `complete`, `</module>` after `</name>`, etc.).

This behavior is present in the current development version of netopeer2 (0.6.8), but was not present prior to 6ab0617.

I've added a unit test in https://github.com/ADTRAN/Netopeer2/commit/429e60dcd439ee84bdec6e74345e4bbc0a5a7da7 that demonstrates the problem.

**Comment by michalvasko** [comment]
Hi,
it should be fixed.

Regards,
Michal

**Comment by alangefe** [comment]
Fixed in 426aba1, included in server-0.6.10.
